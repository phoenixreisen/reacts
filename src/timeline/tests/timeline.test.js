import renderer from 'react-test-renderer';
import TimeLine from '../timeline.r';
import React from "react";

test('Timeline should render left line', () => {
    const component = renderer.create(
        <TimeLine line={'constantly'} position={'left'}>
            <div><strong>Tab I</strong></div>
            <p>Lorem ipsum dolor sit amet, consetetur sadipscing elitr</p>
        </TimeLine>
    );
    const snap = component.toJSON();
    expect(snap).toMatchSnapshot();
});

test('Timeline should render centered line', () => {
    const component = renderer.create(
        <TimeLine line={'separated'} position={'center'}>
            <div><strong>Tab II</strong></div>
            <p>Lorem ipsum dolor sit amet, consetetur sadipscing elitr</p>
        </TimeLine>
    );
    const snap = component.toJSON();
    expect(snap).toMatchSnapshot();
});