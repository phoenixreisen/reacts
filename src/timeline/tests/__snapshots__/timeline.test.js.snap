// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline should render centered line 1`] = `
<section
  className="timeline timeline--static timeline--center"
>
  <article
    className="timeline-box"
  >
    <span
      className="timeline-axis"
      style={
        {
          "transition": "height 500ms ease 0ms",
        }
      }
    />
    <div
      className="timeline-center-box"
    >
      <i
        className="timeline-circle"
      />
      <div
        className="timeline-content timeline-content--left"
      >
        <div>
          <strong>
            Tab II
          </strong>
        </div>
      </div>
      <div
        className="timeline-content timeline-content--right"
      />
    </div>
  </article>
  <article
    className="timeline-box"
  >
    <span
      className="timeline-axis"
      style={
        {
          "transition": "height 500ms ease 500ms",
        }
      }
    />
    <div
      className="timeline-center-box"
    >
      <i
        className="timeline-circle"
      />
      <div
        className="timeline-content timeline-content--left"
      />
      <div
        className="timeline-content timeline-content--right"
      >
        <p>
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr
        </p>
      </div>
    </div>
  </article>
</section>
`;

exports[`Timeline should render left line 1`] = `
<section
  className="timeline  timeline--left"
>
  <span
    className="timeline-axis"
  />
  <article
    className="timeline-box"
  >
    <div
      className="timeline-left-box"
    >
      <i
        className="timeline-circle"
      />
      <div
        className="timeline-content"
      >
        <div>
          <strong>
            Tab I
          </strong>
        </div>
      </div>
    </div>
  </article>
  <article
    className="timeline-box"
  >
    <div
      className="timeline-left-box"
    >
      <i
        className="timeline-circle"
      />
      <div
        className="timeline-content"
      >
        <p>
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr
        </p>
      </div>
    </div>
  </article>
</section>
`;
